src/components/interactive-map.js

```diff
@@ -41,41 +31,86 @@ const propTypes = Object.assign({}, StaticMap.propTypes, {
    */
   onChangeViewport: PropTypes.func,
 
-  /** Enables perspective control event handling */
-  perspectiveEnabled: PropTypes.bool,
-
+  /** Enables control event handling */
+  // Scroll to zoom
+  scrollZoomEnabled: PropTypes.bool,
+  // Drag to pan
+  dragPanEnabled: PropTypes.bool,
+  // Drag to rotate
+  dragRotateEnabled: PropTypes.bool,
+  // Double click to zoom
+  doubleClickZoomEnabled: PropTypes.bool,
+  // Pinch to zoom / rotate
+  touchZoomRotateEnabled: PropTypes.bool,
+
+ /**
+    * Called when the map is hovered over.
+    * @callback
+    * @param {Object} event - The mouse event.
+    * @param {[Number, Number]} event.lngLat - The coordinates of the pointer
+    * @param {Array} event.features - The features under the pointer, using Mapbox's
+    * queryRenderedFeatures API:
+    * https://www.mapbox.com/mapbox-gl-js/api/#Map#queryRenderedFeatures
+    * To make a layer interactive, set the `interactive` property in the
+    * layer style to `true`. See Mapbox's style spec
+    * https://www.mapbox.com/mapbox-gl-style-spec/#layer-interactive
+    */
+  onHover: PropTypes.func,
   /**
-    * Is the component currently being dragged. This is used to show/hide the
-    * drag cursor. Also used as an optimization in some overlays by preventing
-    * rendering while dragging.
+    * Called when the map is clicked.
+    * @callback
+    * @param {Object} event - The mouse event.
+    * @param {[Number, Number]} event.lngLat - The coordinates of the pointer
+    * @param {Array} event.features - The features under the pointer, using Mapbox's
+    * queryRenderedFeatures API:
+    * https://www.mapbox.com/mapbox-gl-js/api/#Map#queryRenderedFeatures
+    * To make a layer interactive, set the `interactive` property in the
+    * layer style to `true`. See Mapbox's style spec
+    * https://www.mapbox.com/mapbox-gl-style-spec/#layer-interactive
     */
-  isHovering: PropTypes.bool,
-  isDragging: PropTypes.bool,
+  onClick: PropTypes.func,
+
+  /** Radius to detect features around a clicked point. Defaults to 0. */
+  clickRadius: PropTypes.number,
+
+  /** Accessor that returns a cursor style to show interactive state */
+  getCursor: PropTypes.func,
 
   /** Advanced features */
   // Contraints for displaying the map. If not met, then the map is hidden.
-  displayConstraints: PropTypes.object.isRequired,
+  // Experimental! May be changed in minor version updates.
+  visibilityConstraints: PropTypes.object.isRequired,
   // A map control instance to replace the default map controls
   // The object must expose one property: `events` as an array of subscribed
```

Comment:
Would love to see this moved to markdown and expanded upon. Semantics are not yet fully clear to me.